sbtc-core/src/signer/mod.rs

```diff
@@ -0,0 +1,119 @@
+/// An sBTC signer configuration
+pub mod config;
+
+use std::collections::HashMap;
+
+use crate::signer::config::Config;
+use crate::SBTCError;
+use bitcoin::{Network, PrivateKey, PublicKey, Transaction as BitcoinTransaction};
+use p256k1::ecdsa;
+use url::Url;
+use wsts::{bip340::SchnorrProof, common::Signature};
+
+#[derive(Default, Clone, Debug)]
+pub struct SignersData {
+    pub signers: HashMap<u32, ecdsa::PublicKey>,
+    pub vote_shares: HashMap<u32, ecdsa::PublicKey>,
+}
+
+/// Coordinator algorithm trait
+pub trait CoordinatorAlgorithm {
+    fn generate_sbtc_wallet_public_key(&self) -> Result<PublicKey, SBTCError>;
+    fn run_signing_round(
+        &self,
+        _tx: BitcoinTransaction,
+    ) -> Result<(Signature, SchnorrProof), SBTCError>;
+}
+
+/// Signer algorithm trait
+pub trait SignerAlgorithm: CoordinatorAlgorithm {
```

Comment:
Why does a signer algorithm also have to be a coordinator algorithm?

Reply:
I also believe if there's no reason to split these two I'd rather keep them in a single trait. Also, can't say I know Rust guidelines for trait names too well but something like `Coordinate` and `Sign` for some reason sounds better. A very small preference for this so please disregard if you don't agree.

Reply:
Every signer also needs to be able to operate as a coordinator. But the logic behind coordination is independent of the logic for the actual math behind the signing.

Reply:
Ah okay got it. In some sense both coordinating and signing is a part of the same process, but maybe it will be easier to understand and present if separated into two distinct traits?

Reply:
Oh I am misunderstanding your feedback. Yes good point, they should be sep traits, not inheriting from each other. Fixed!